### PR TITLE
Tree widget: matching instances count loading indicator

### PR DESCRIPTION
**Pull request by jasdom**
Closes https://github.com/iTwin/viewer-components-react/issues/909.

### CODE REVIEW

**Review by grigasp — COMMENTED** [review]
Looks like there's no delay - how does this look when the count gets calculated quickly?

**Comment by jasdom** [comment]
> Looks like there's no delay - how does this look when the count gets calculated quickly?

As discussed added a delay before showing the indicator.

File: change/@itwin-tree-widget-react-139500ee-01e4-4ff2-9d52-3020e9ec60b7.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (grigasp) [inline change/@itwin-tree-widget-react-139500ee-01e4-4ff2-9d52-3020e9ec60b7.json:2]:
```suggestion
  "type": "patch",
```

**Review by grigasp — APPROVED** [review]

File: packages/itwin/tree-widget/src/components/trees/stateless/common/UseHierarchyFiltering.tsx [hunk 2]
```diff
@@ -116,10 +118,12 @@ interface MatchingInstancesCountProps {
 }
 
 function MatchingInstancesCount({ filter, defaultHierarchyLevelSizeLimit, hierarchyLevelDetails }: MatchingInstancesCountProps) {
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (saskliutas) [inline packages/itwin/tree-widget/src/components/trees/stateless/common/UseHierarchyFiltering.tsx:121]: Looks like `useDebouncedAsyncValue` returns `inProgress` flag https://github.com/iTwin/appui/blob/5995ba38b3c053d8d2551a7ef460e33b0e0fc466/ui/components-react/src/components-react/common/UseDebouncedAsyncValue.ts#L80C27-L80C38

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: packages/itwin/tree-widget/src/components/trees/stateless/common/UseHierarchyFiltering.tsx (hunk 2 above)
Comment (jasdom) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/UseHierarchyFiltering.tsx]: Read the entire description but somehow missed the flag. Changed.

**Review by saskliutas — APPROVED** [review]

**Pull request merged**